nixos/modules/services/networking/nftables.nix

```diff
@@ -144,7 +150,12 @@ in
           checkPhase = lib.optionalString cfg.checkRuleset ''
             cp $out ruleset.conf
             ${cfg.preCheckRuleset}
-            export NIX_REDIRECTS=/etc/protocols=${pkgs.buildPackages.iana-etc}/etc/protocols:/etc/services=${pkgs.buildPackages.iana-etc}/etc/services
+            export NIX_REDIRECTS=${concatStringsSep ":"
+              [ "/etc/protocols=${pkgs.buildPackages.iana-etc}/etc/protocols"
+                "/etc/services=${pkgs.buildPackages.iana-etc}/etc/services"
+                "/etc/hosts=${config.environment.etc.hosts.source}"
```

Comment:
Does non-library redirection actually work with this? I was under the impression `NIX_REDIRECTS` were wholly intended for linking redirects.

Reply:
Yeah, it's just hooking the system call wrappers (`open()`, etc.) in libc to rewrite a path.